File: Practice/33/Python/zadanie_33.py
def create(arr_len, arr_1st_el=0, arr_step=0):
    return [ i for i in range(arr_1st_el, arr_1st_el + arr_len*arr_step, arr_step) ] \
        if arr_step else [arr_1st_el]*arr_len

    #if not arifm_step: return [arr_first_elem]*arr_len
    #return [ i for i in range(arr_first_elem, arr_first_elem + arr_len*arifm_step, arifm_step) ]

    #return [arr_first_elem + i*arifm_step for i in range(arr_len)]

def sort(arr): # намного быстрее просто использовать встроенную функцию сортировки: return sorted(arr)
    for i in range(len(arr)):
        for j in range(i, 0,-1):
            if arr[j - 1] <= arr[j]: break
            arr[j - 1], arr[j]=arr[j], arr[j - 1]
    return arr

_prn=print
def print(arr):
    _prn(arr)
    return arr

print(sort(create(arr_len=int(input()), arr_1st_el=int(input()), arr_step=int(input()))))

#for i in range(1000000):
#    create(8,4,-1)

File: Practice/33/C++/Zadanie_33/Zadanie_33.cpp
#include <iostream>
using namespace std;

//длину массива, начальный член и шаг арифметической прогрессии
int* create(int arr_len, int arr_first_elem = 0, int arifm_step = 0) {

    int* new_array = new int[arr_len];

    // Инициализация массива
    for (int i = 0; i < arr_len; i++) {
        new_array[i] = arr_first_elem;
        arr_first_elem = arr_first_elem + arifm_step;
    }
    return new_array;
}

int* sort(int* arr, int arr_len) {
    for (int i = 1; i < arr_len; i++)
        for (int j = i; j > 0 && arr[j - 1] > arr[j]; j--) // пока j>0 и элемент j-1 > j, x-массив int
            swap(arr[j - 1], arr[j]);                      // меняем местами элементы j и j-1
    return arr;
}

int* print(int* arr, int arr_len) {
    cout << "[";
    for (int i = 0; i < arr_len; i++) {
        cout << arr[i] << ((i < arr_len - 1) ? ", " : "");
    }
    cout << "]";
    return arr;
}

int main()
{
    int arr_len, arr_first_elem, arifm_step;
    cin >> arr_len >> arr_first_elem >> arifm_step;
    int* arr = create(arr_len, arr_first_elem, arifm_step);
    sort(arr, arr_len);
    print(arr, arr_len);
    delete[] arr;
    return 0;
}
